packages/rspack/src/builtin-plugin/css-extract/hmr/hotModuleReplacement.ts

```diff
@@ -246,12 +242,7 @@ function isUrlRequest(url) {
 	return true;
 }
 
-/**
- * @param {TODO} moduleId
- * @param {TODO} options
- * @returns {TODO}
- */
-module.exports = function (moduleId, options) {
+module.exports = function (moduleId: string, options: Record<string, any>) {
```

Comment:
can this function be replaced with `export default`?

Reply:
It seems using esm export will cause some test cases to fail

```
│   ● hotSnapshot › hotSnapshot/css/recovery › step  should pass
│     TypeError: __webpack_require__(...) is not a function
│
│       at Module../index.css?8047 (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:274:216)
│       at __webpack_require__ (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:311:21)
│       at fn (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:432:10)
│       at Module../index.js (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:26:67)
│       at __webpack_require__ (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:311:21)
│       at eval (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:1470:27)
│       at eval (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:1472:3)
│       at eval (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:1474:11)
│           at eval (<anonymous>)
│       at dist/runner/runner/web/jsdom.js:179:29
│   ● hotSnapshot › hotSnapshot/css/recovery-cacheable › step  should pass
│     TypeError: __webpack_require__(...) is not a function
│
│       at Module../change.css?a8a3 (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:292:21…
│       at __webpack_require__ (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:344:21)
│       at fn (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:465:10)
│       at Module../change.js (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:26:68)
│       at __webpack_require__ (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:344:21)
│       at fn (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:465:10)
│       at Module../index.js (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:34:64)
│       at __webpack_require__ (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:344:21)
│       at eval (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:1503:27)
│       at eval (eval at <anonymous> (dist/runner/runner/web/jsdom.js:179:29), <anonymous>:1505:3)
│ Test Suites: 1 failed, 1 total
│ Tests:       2 failed, 2 skipped, 180 passed, 184 total
│ Snapshots:   0 total
│ Time:        6.449 s
```

Reply:
If I use esm export, I think I need to update jest config. Would you like that?
 
```
 FAIL  css-extract/HMR.test.js
  ● Test suite failed to run

    Jest encountered an unexpected token

    Jest failed to parse a file. This happens e.g. when your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax.

    Out of the box Jest supports Babel, which will be used to transform your files into valid JS based on your Babel configuration.

    By default "node_modules" folder is ignored by transformers.

    Here's what you can do:
     • If you are trying to use ECMAScript Modules, see https://jestjs.io/docs/ecmascript-modules for how to enable it.
     • If you are trying to use TypeScript, see https://jestjs.io/docs/getting-started#using-typescript
     • To have some of your "node_modules" files transformed, you can specify a custom "transformIgnorePatterns" in your config.
     • If you need a custom transformation specify a "transform" option in your config.
     • If you simply want to mock your non-JS modules (e.g. binary assets) you can stub them out with the "moduleNameMapper" config option.

    You'll find more details and examples of these config options in the docs:
    https://jestjs.io/docs/configuration
    For information about custom transformations, see:
    https://jestjs.io/docs/code-transformation

    Details:

    /home/wuxiaoyun/dev/rspack/tests/plugin-test/css-extract/HMR.test.js:7
    import hotModuleReplacement from "../../../packages/rspack/dist/builtin-plugin/css-extract/hmr/hotModuleReplacement" ;
    ^^^^^^

    SyntaxError: Cannot use import statement outside a module

      at Runtime.createScriptFromCode (../../node_modules/.pnpm/jest-runtime@29.7.0/node_modules/jest-runtime/build/index.js:1505:14)
```